lib/beaker/host/unix/pkg.rb

```diff
@@ -120,6 +120,34 @@ def install_package(name, cmdline_args = '', version = nil, opts = {})
     end
   end
 
+  # Extract RPM command's proxy options from URL
+  #
+  # @param [Url] url A URL of form http://host:port
+  #
+  def extract_rpm_proxy_options(url)
```

Comment:
I know that you do guard against `opts[:package_proxy]` being undefined in `install_package_with_rpm`, but it would be even safer if you guarded against anyone passing nil in by accident.

If you consider this dealt with in the catching of the `url.match` failure, then I'm down with that choice.  If that's the case, it might be better to put quotes around the `#{url}` part of the exception, so that it's easier to see it's nil rather than wondering if there might be a newline in there or something.